azkaban-common/src/main/java/azkaban/Constants.java

```diff
@@ -110,6 +110,17 @@
     public static final String AZKABAN_KERBEROS_PRINCIPAL = "azkaban.kerberos.principal";
     public static final String AZKABAN_KEYTAB_PATH = "azkaban.keytab.path";
     public static final String PROJECT_TEMP_DIR = "project.temp.dir";
+
+    /*
+     * The max number of artifacts retained per project.
+     * Example: If this config is set to 3, then we have at latest 3 versions saved in storage.
+     * Constraints:
+     * - Has to be Integer
+     * - Negative values are not allowed and will crash the app.
+     * - A value of 0 implies all artifacts will be stored [Default]
```

Comment:
@HappyRay 
I would prefer to stick to `all (0)` as the default for now since it is in line with the current implementation. Therefore any new build doesn't suddenly change the behavior of the app. It's only when you turn on this knob, you see a behavior change.

@reallocf 
I like the idea! It is much more explicit. However, I'm a bit reluctant to introduce that complexity in an almost non-user facing, and probably seldom used config variable. This keeps it very crisp. Less validation. Less code.
The same goes for introducing another boolean config variable. Thoughts?